Sources/mpc-core-kit-swift/Helper.swift

```diff
@@ -79,3 +78,32 @@ public extension Web3AuthNetwork {
         return convertWeb3AuthNetworkToTorusNetWork(network: self)
     }
 }
+
+public func parseToken(jwtToken jwt: String) -> [String: Any] {
+  let segments = jwt.components(separatedBy: ".")
+  return decodeJWTPart(segments[1]) ?? [:]
+}
+
+func base64UrlDecode(_ value: String) -> Data? {
+  var base64 = value
+    .replacingOccurrences(of: "-", with: "+")
+    .replacingOccurrences(of: "_", with: "/")
+
+  let length = Double(base64.lengthOfBytes(using: String.Encoding.utf8))
+  let requiredLength = 4 * ceil(length / 4.0)
+  let paddingLength = requiredLength - length
+  if paddingLength > 0 {
+    let padding = "".padding(toLength: Int(paddingLength), withPad: "=", startingAt: 0)
+    base64 = base64 + padding
+  }
+  return Data(base64Encoded: base64, options: .ignoreUnknownCharacters)
+}
+
+func decodeJWTPart(_ value: String) -> [String: Any]? {
+  guard let bodyData = base64UrlDecode(value),
+    let json = try? JSONSerialization.jsonObject(with: bodyData, options: []), let payload = json as? [String: Any] else {
+      return nil
+  }
+
+  return payload
+}
```

Comment:
This should throw if the jwt is not decodable, not return nil.
```
func decodeJWTPart(_ value: String) throws -> [String: Any] {
...
}
```

Could you not just use the jwtDecode package instead?

i.e
```
 return try decode(jwt: idToken)
```